### DISCUSSION TITLE
Parse sidewalks (walkways) of type multipolygone

### DISCUSSION THREAD

**Discussion by oliverpapst1** [CommonRoad/commonroad-scenario-designer#10] [Q&A] [answered]
Hello all!

So far I noticed that when parsing sidewalk information, only relations defined by 2 way tags, namely with roles left and right are parsed.

Is there also an option to parse sidewalk information from multipolygones defined by multiple way tags with role outer?

Thanks a lot.

Best,
Oliver

**Comment by smaierhofer** [comment]
To which conversion are you referring, Lanelet2 to CommonRoad or OpenStreetMap to CommonRoad?

**Reply by oliverpapst1** [reply]
LL2 to CR, i.e. this part of the code:

```
        for way_rel in self.xml.xpath("//relation/tag[@v='lanelet' and @k='type']/.."):
            try:
                left_way_id = way_rel.xpath("./member[@type='way' and @role='left']/@ref")[0]
                right_way_id = way_rel.xpath("./member[@type='way' and @role='right']/@ref")[0]
```

If I understood correctly, walkway multipolygons with ways of role 'outer' will not be parsed by the converter.

**Reply by smaierhofer** [reply, marked as answer]
No, this is currently not supported. 
However, CommoRoad could probably represent it via areas.

**Reply by oliverpapst1** [reply]
Hmm I see, thank you for your help and suggestion Sebastian.

Best,
Oliver

**Reply by smaierhofer** [reply]
Do you have an example map where this is used then we could integrate it in the future?

**Reply by oliverpapst1** [reply]
I am currently trying location4.osm of the inD dataset.

https://levelxdata.com/ind-dataset/#download

There should be some examples included of sidewalks defined as multiploygons.
